FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,46 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard let self = self else { return }
+
+			guard let (data, response) = try? result.get() else {
+				completion(.failure(RemoteFeedLoader.Error.connectivity))
+				return
+			}
+
+			completion(self.map(response, withData: data))
+		}
+	}
+
+	// MARK: - Helper Methods
+
+	private func map(_ response: HTTPURLResponse, withData data: Data) -> FeedLoader.Result {
+		let decoder = JSONDecoder()
+
+		if response.statusCode == 200, let images = try? decoder.decode(Items.self, from: data) {
+			return .success(images.feedImages)
+		} else {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
+	}
+
+	private struct Items: Codable {
+		private var items: [FeedItem]
+
+		var feedImages: [FeedImage] {
+			return items.map { $0.feedImage }
+		}
+	}
+
+	private struct FeedItem: Codable {
+		private var image_id: UUID
+		private var image_desc: String?
+		private var image_loc: String?
+		private var image_url: URL
```

Comment:
The properties are never mutated so they can be declared as let instead of var:

```suggestion
		private let image_id: UUID
		private let image_desc: String?
		private let image_loc: String?
		private let image_url: URL
```